instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/TracingRequestWrapper.java

```diff
@@ -6,18 +6,32 @@
 package io.opentelemetry.instrumentation.awslambda.v1_0;
 
 import io.opentelemetry.sdk.OpenTelemetrySdk;
+import java.util.function.BiFunction;
 
 /**
  * Wrapper for {@link TracingRequestHandler}. Allows for wrapping a regular lambda, not proxied
  * through API Gateway. Therefore, HTTP headers propagation is not supported.
  */
 public class TracingRequestWrapper extends TracingRequestWrapperBase<Object, Object> {
   public TracingRequestWrapper() {
-    super();
+    super(TracingRequestWrapper::map);
   }
 
   // Visible for testing
-  TracingRequestWrapper(OpenTelemetrySdk openTelemetrySdk, WrappedLambda wrappedLambda) {
-    super(openTelemetrySdk, wrappedLambda);
+  TracingRequestWrapper(
+      OpenTelemetrySdk openTelemetrySdk,
+      WrappedLambda wrappedLambda,
+      BiFunction<Object, Class, Object> mapper) {
+    super(openTelemetrySdk, wrappedLambda, mapper);
+  }
+
+  // Visible for testing
+  static Object map(Object jsonMap, Class clazz) {
```

Comment:
I'm surprised this non-API Gateway class (so not populating HTTP attrs) is mapping, does it need to?

Reply:
With our wrapper requesting Object input type, AWS will (out of my testing):
- invoke with value in case of primitive values 
- invoke with Map<String, Object> in case of custom types
So in order to support pre-mapped custom types we need to map the Map into appropriate class.